python_utils.py

```diff
@@ -393,19 +393,18 @@ def url_retrieve(source_url, filename=None):
     Returns:
         urlretrieve. The 'urlretrieve' object.
     """
-    context = ssl.create_default_context(cafile=certifi.where())
     try:
         import urllib.request as urlrequest
     except ImportError:
         import urllib as urlrequest
+
         # Change the User-Agent to prevent servers from blocking requests.
         # See https://support.cloudflare.com/hc/en-us/articles/360029779472-Troubleshooting-Cloudflare-1XXX-errors#error1010. # pylint: disable=line-too-long
-        urlrequest.URLopener.version = (
-            'Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) '
-            'Gecko/20100101 Firefox/47.0'
-        )
-    return urlrequest.urlretrieve(
-        source_url, filename=filename, context=context)
+    urlrequest.URLopener.version = (
```

Comment:
No, it shouldn't, it is needed for both py2 ad py3. Theoretically, we can remove this file already since it was needed for the compatibility, meaning that we can remove individual parts and only keep the py3 related ones, but since this can also be done later we decided to keep this file to reduce the work.

Reply:
OK, I'm fine leaving the try/except in. But why is the block of code above moving from the "except" to the main body? I still don't follow the reasoning.

Reply:
Because it is needed for both libraries. It was a mistake to not set it before.